common/src/main/java/com/wynntils/models/abilities/type/AbilityTreeSkillNode.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.abilities.type;
+
+import com.wynntils.core.text.PartStyle;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.utils.mc.LoreUtils;
+import com.wynntils.utils.type.IterationDecision;
+import com.wynntils.utils.type.Pair;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+
+public record AbilityTreeSkillNode(
+        String name,
+        String formattedName,
+        List<String> description,
+        ItemInformation itemInformation,
+        int cost,
+        List<String> blocks,
+        String requiredAbility,
+        ArchetypeRequirement requiredArchetype,
+        String archetype,
+        AbilityTreeLocation location,
+        List<String> connections) {
+    private static final Pattern NODE_NAME_PATTERN = Pattern.compile("§.(Unlock )?§l(.+)(§r§. ability)?");
+    private static final Pattern NODE_POINT_COST_PATTERN = Pattern.compile("§.. §7Ability Points: §f(\\d+)");
+    private static final Pattern NODE_BLOCKS_ABILITY_PATTERN = Pattern.compile("§c- §7(.+)");
+    private static final Pattern NODE_REQUIRED_ABILITY_PATTERN = Pattern.compile("§.. §7Required Ability: §f(.+)");
+    private static final Pattern NODE_REQUIRED_ARCHETYPE_PATTERN =
+            Pattern.compile("§.. §7Min (.+) Archetype: §c(\\d+)§7/(\\d+)");
+    private static final Pattern NODE_ARCHETYPE_PATTERN = Pattern.compile("§.§l(.+) Archetype");
+    private static final Pattern NODE_BLOCKED = Pattern.compile("§cBlocked by another ability");
+    private static final Pattern NODE_UNLOCKED = Pattern.compile("§eYou already unlocked this ability");
+
+    public static Pair<AbilityTreeSkillNode, AbilityTreeNodeState> parseNodeFromItem(
```

Comment:
When the classes start getting this much complex logic I do not think they belong in `type`. Either move the parsing to another class, like AbilityModel, or a new AbilityParser, or whatever, or move the class to just abilities, not abilities.type.